### ISSUE TITLE
Search endpoint to perform elasticsearch query

### ISSUE THREAD

**Issue by chrisaballard** [climatepolicyradar/cpr-issues#61]

**Comment by chrisaballard** [comment]
@eurolife here is the proposed api path and response. Does this work for you?

**Comment by eurolife** [comment]
@chrisaballard looks okay to me, what is the meaning of the page number value? is that the number of total pages in the document, or?

**Comment by chrisaballard** [comment]
```pageNumber``` would return the page number that the matches are found in. At this point, you won't know the total number of pages, but you would then call the ```policy/{policy_id}/text/``` path to return the full text for that page, and highlight the matched text once the user clicks on the match. The latter call would return the total number of pages that you can then use for pagination. Make sense?

**Comment by chrisaballard** [comment]
> @chrisaballard looks okay to me, what is the meaning of the page number value? is that the page number where the text passage can be found? Also can I get info in the response to help with pagination?

pageNumber would return the page number that the matches are found in. At this point, you won't know the total number of pages, but you would then call the policy/{policy_id}/text/ path to return the full text for that page, and highlight the matched text once the user clicks on the match. The latter call would return the total number of pages that you can then use for pagination. Make sense?

**Comment by eurolife** [comment]
@chrisaballard Yes that makes sense to me! I should be able to highlight the matched selection of text in a nice way.

**Comment by kdutia** [comment]
There seems to be no easy way to return the total number of results for an `aggregate` Elasticsearch query, except for returning a high number (which is expensive) and paginating it on our server. 

- https://stackoverflow.com/questions/61106956/how-to-include-all-docs-in-elasticsearch-aggregation-and-avoid-sum-other-doc-cou
- https://discuss.elastic.co/t/sum-other-doc-count-higher-than-total-docs/140490

**Comment by kdutia** [comment]
Query is based around [top hits aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-top-hits-aggregation.html#_field_collapse_example)

**Comment by eurolife** [comment]
@kdutia I've tried to query this and it works with just one country code but not multiple country codes. It returns 0 results with multiple country codes.
